src/components/markdown/markdown.tsx

```diff
@@ -15,13 +15,33 @@ import remarkRaw from 'rehype-raw';
 import rehype2react from 'rehype-react';
 import { isValidElementType } from 'react-is';
 import { root as mdRoot } from 'mdast-builder';
-import mdxComponentMap from '../../utils/mdx-component-mapper';
+import mdxComponentMap, {
+  MarkdownComponentMap
+} from '../../utils/mdx-component-mapper';
 import indentNormalizer from '../../utils/indent-normalizer';
 import Notes from '../notes';
 import { ListItem } from '../../index';
 import { Appear } from '../appear';
 
-export const Markdown = React.forwardRef(
+type MdComponentProps = { [key: string]: any };
```

Comment:
This file was the hardest to convert; I left a few items as `any` or `unknown`.

Reply:
It could be a serializable object, a React node, a function. But since it's a prop it could also be a falsy value. I think this makes sense.